pkg/operator/controller/ingress/controller.go

```diff
@@ -373,6 +373,15 @@ func setDefaultPublishingStrategy(ic *operatorv1.IngressController, infraConfig
 		ic.Status.EndpointPublishingStrategy = effectiveStrategy
 		return true
 	}
+
+	// Detect changes to GCP LB provider parameters, which is something we can safely roll out.
+	statusLB := ic.Status.EndpointPublishingStrategy.LoadBalancer
+	specLB := effectiveStrategy.LoadBalancer
+	if specLB != nil && statusLB != nil && specLB.ProviderParameters != nil && statusLB.ProviderParameters != nil &&
+		specLB.ProviderParameters.GCP != statusLB.ProviderParameters.GCP {
+		ic.Status.EndpointPublishingStrategy.LoadBalancer.ProviderParameters.GCP = effectiveStrategy.LoadBalancer.ProviderParameters.GCP
```

Comment:
It might be a good time to remove the return value, since it isn't used by the caller.  If we anticipate it to be used in the future, you might want to return true here.

Reply:
In the future we might use the return value from `setDefaultPublishingStrategy` to detect if any changes were made to `ic.Status`, so for now I'll add a return value here (surprisingly this was also overlooked in the reverted mutable-scope PR https://github.com/openshift/cluster-ingress-operator/pull/472, so nice catch!).

Reply:
Fixed.